### ISSUE TITLE
Randomly changing tracking device

### ISSUE THREAD

**Issue by jspeckman** [gcobb321/icloud3_v3#201]
HI,

I'm having a problem which I've not been able to find an answer to.
I'm running the latest icloud3 with the latest stable home assistant.
My tracked device will randomly change from my current iphone to an older iphone.
I removed the older iphone from my apple account, however icloud3 still insists on switching back to at some point after changing the tracked device back to my current iphone.
I've looked through the config files, but have not found the old iphone listed anywhere.
I don't know if this is a bug or I'm just doing something wrong.
Also, my current and old iphone are/were both named iPhone.

Thanks,
Jason

**Comment by gcobb321** [comment]
Go to Configure Settings > Update Devices and select the iCloud3 device. Change the FamShr device to the one you want to track from the drop down list of devices in your Apple account. 

If you still have the old phone, rename it on the phone to make it easier to identify it in the list. If you don’t, select the correct one from the list using the device type.

**Comment by jspeckman** [comment]
That's what I've been doing.
After a few days, it gets changed back.  When I notice it (no longer getting tracking info), I go back in and change it again.

I do still have the old phone, I've removed it from my apple account, but it still shows up in the device drop down list (it does not show up on my list of devices on the apple site, or in my phone), and gets selected within a day or two of manually choosing the correct one.

As far as identifying the device, while they are both named iPhone, the model info is shown in parenthesis in the drop down list, so it is easy to tell which is which.

**Comment by gcobb321** [comment]
Stage 3 and 4 in the Event Log during iCloud3 start up show the devices read from your iCloud Family Sharing list. Are both phones listed (old and new)? Does it show any messages related to duplicate devices w with the same name in the Event Log?

**Comment by jspeckman** [comment]
Both devices were listed in the event log at stage 3.

It seems that while I had removed the device from my apple account, it was still listed as a device in FindMy.
After removing the device from FindMy, it continued to show up in the stage 3 list.  I figured it might be a caching issue, so I stopped Home Assistant for about an hour.  On restart, only the current device was listed under stage 3, so it appears to be fixed now.

Thanks for the help and pointers on where to look to troubleshoot.
